siwe/siwe.py

```diff
@@ -142,7 +142,8 @@ class SiweMessage(BaseModel):
         None, min_items=1
     )  # List of information or references to information the user wishes to have resolved as part of authentication by the relying party. They are expressed as RFC 3986 URIs separated by `\n- `.
 
-    def __init__(self, message: Union[str, dict], abnf: bool = True):
+    @classmethod
+    def from_message(cls, message: Union[str, dict], abnf: bool = True) -> "SiweMessage":
```

Comment:
if this is being done, I think `from_message` should only works on a string. Passing a dict does not do much except just call the pydantic class on the dict. This can just be done by doing `SiweMessage(**dict)`. This also removes type assertions and makes this code much cleaner